### PR TITLE
[UNTIL-10876] Scan engine detekt proguard setup

### PR DESCRIPTION
**Pull request by shekar-allam**

### CODE REVIEW

File: scan-engine/src/main/java/de/tillhub/scanengine/default/ui/GoogleScanningViewModel.kt [hunk 1]
```diff
@@ -45,21 +45,19 @@ private class QRImageAnalyzer(
...
         }
     }
 }
 
-sealed class ScanningState {
-    object Idle : ScanningState()
-    data class CodeScanned(val barcode: String) : ScanningState()
+internal sealed class ScanningState {
+    data object Idle : ScanningState()
+    class CodeScanned(val barcode: String) : ScanningState()
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/default/ui/GoogleScanningViewModel.kt:62]: This is ok to be data class if it's used internal. Only classes that are exposed outside of the sdk should not be data class

File: scan-engine/src/main/java/de/tillhub/scanengine/helper/BundleExtension.kt [hunk 2]
```diff
@@ -1,16 +1,10 @@
 package de.tillhub.scanengine.helper
 
-import android.content.Intent
 import android.os.Build
 import android.os.Bundle
 import java.io.Serializable
 
-inline fun <reified T : Serializable> Bundle.serializable(key: String): T? = when {
+internal inline fun <reified T : Serializable> Bundle.serializable(key: String): T? = when {
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/helper/BundleExtension.kt:7]: Update code-ktx and use BundleCompact they support now getSerializable and you can delete then our BundleExtension

File: scan-engine/src/main/java/de/tillhub/scanengine/sunmi/SunmiScanner.kt [hunk 3]
```diff
@@ -36,7 +36,7 @@ class SunmiScanner(
     override fun observeScannerResults(): StateFlow<ScanEvent> = mutableScanEvents
 
     sealed class ScanCodeType {
-        data class Type(val code: String) : ScanCodeType()
+        class Type(val code: String) : ScanCodeType()
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/sunmi/SunmiScanner.kt:39]: same here, no reason to not keep as data class.

File: scan-engine/src/main/java/de/tillhub/scanengine/sunmi/SunmiScannerActivityContract.kt [hunk 4]
```diff
@@ -66,7 +66,7 @@ class SunmiScannerActivityContract : ActivityResultContract<Unit, List<ScanEvent
         else -> SunmiScanner.ScanCodeType.Type(this)
     }
 
-    private data class ScanCode(
+    private class ScanCode(
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/sunmi/SunmiScannerActivityContract.kt:69]: keep it as data class

**Review by djordjeh — CHANGES_REQUESTED** [review]

**Review by SloInfinity — APPROVED** [review]

**Pull request merged**